state-chain/custom-rpc/src/lib.rs

```diff
@@ -451,6 +465,12 @@ pub trait CustomApi {
 	#[subscription(name = "subscribe_prewitness_swaps", item = BlockUpdate<RpcPrewitnessedSwap>)]
 	fn cf_subscribe_prewitness_swaps(&self, base_asset: Asset, quote_asset: Asset, side: Order);
 
+	// Subscribe to a stream that on every block produces a list of all scheduled/pending
+	// swaps in the base_asset/quote_asset pool, including any "implicit" half-swaps (as a
+	// part of a swap involving two pools)
+	#[subscription(name = "subscribe_scheduled_swaps", item = SwapResponse)]
+	fn cf_subscribe_scheduled_swaps(&self, base_asset: Asset, quote_asset: Asset);
```

Comment:
This rpc call doesn't seem to have any checks on quote_asset? It should atleast fail, if it is not USDC, ideally I'd suggest just checking if the pool exists in `cf_scheduled_swaps`, and returning an error if it doesn't (In the pools pallet there is an error for this)

Reply:
I reused `cf_pool_info` for this (creating a separate method returning true/false didn't seem worth it)